app/routes/items/detail.js

```diff
@@ -36,20 +46,87 @@ export default AuthorizeRoute.extend({
     var previousRoutes = this.router.router.currentHandlerInfos;
     var previousRoute = previousRoutes && previousRoutes.pop();
     var path = "items.index";
-    if(previousRoute) {
+    if (previousRoute) {
       var routeName = previousRoute.name;
-      if(routeName.indexOf("items") === 0) {
+      if (routeName.indexOf("items") === 0) {
         path = this.get("itemBackLinkPath") || path;
-      } else if(routeName.indexOf("items") > -1 || routeName === "orders.detail"){
+      } else if (
+        routeName.indexOf("items") > -1 ||
+        routeName === "orders.detail"
+      ) {
         path = routeName;
       }
     }
     this.set("itemBackLinkPath", path);
   },
 
-  setupController(controller, model){
+  setupController(controller, model) {
     this._super(controller, model);
-    controller.set('callOrderObserver', false);
-    controller.set('backLinkPath', this.get('itemBackLinkPath'));
+    controller.set("showSetList", false);
+    controller.set("callOrderObserver", false);
+    controller.set("backLinkPath", this.get("itemBackLinkPath"));
+    controller.set("active", true);
+  },
+
+  resetController(controller, isExiting) {
+    if (isExiting) {
+      controller.set("active", false);
+    }
+  },
+
+  // --- Helpers
+
+  /**
+   * Loads all the images of an item (cached)
+   *
+   * @param {Item} item the selected item
+   */
+  preloadImages(item) {
+    const ids = item.getWithDefault("imageIds", []);
+    return Ember.RSVP.all(ids.map(id => this.loadImage(id)));
+  },
+
+  /**
+   * Loads an image if not available
+   *
+   * @param {String} id the image id
+   * @returns {Image} the image record
+   */
+  async loadImage(id) {
+    const cachedImg = this.store.peekRecord("image", id);
+    if (cachedImg) {
+      return cachedImg;
+    }
+    return this.store.findRecord("image", id, { reload: true });
+  },
+
+  /**
+   * Loads an item with its images
+   * Shorthand for loadItem(id, { loadImages: true })
+   *
+   * @param {String} id the item id
+   * @returns {Item} the item record
+   */
+  loadItemWithImages(id) {
+    return this.loadItem(id, { loadImages: true });
+  },
+
+  /**
+   * Loads an item
+   * Uses the reload:false option to prevent background loads
+   *
+   * @param {String} id the item id
+   * @param {Object} opts load options
+   * @param {Boolean} opts.loadImages if set to true, will also load the images
+   * @returns {Item} the item record
+   */
+  async loadItem(id, opts = {}) {
+    const { loadImages = false } = opts;
+
+    const it = await this.store.findRecord("item", id, { reload: true });
```

Comment:
What does `it` refers to ?

Reply:
oh, it's meant as a shorter`item` , I can replace it if unclear.

I usually use it in loops  `items.forEach(it => { ... })` , but I guess it doesn't make too much sense here

Reply:
Updated 👌